### PR TITLE
Fixes #1002, updating the EC2 workerpool type for ROSA HCP cluster nodes

### LINKED ISSUE #1002: Change worker pool instance type from m5.4xlarge to m5.2xlarge
```[tasklist]
### Tasks
- [x] update the workerpool type in all places both IaC and Docs
- [x] run the tests to make sure deployment goes through as expected
- [x] analyze the run results for any issues
- [x] Update Keycloak docs - https://github.com/keycloak/keycloak/blob/316e00bb98fd33bac0af5fdbaf7caac4481bee82/docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc?plain=1#L130
```

### PR DESCRIPTION
**Pull request by kami619**
Fixes #1002

### CODE REVIEW

File: doc/benchmark/modules/ROOT/pages/report/rosa-benchmark-key-results.adoc [hunk 1]
```diff
@@ -53,7 +53,7 @@ Each test ran for 10 minutes.
 +
 [source,bash,subs="+quotes"]
 ----
-rosa edit machinepool -c  **<clustername>** --min-replicas 3 --autorepair scaling
+rosa edit machinepool -c  **<clustername>** --min-replicas 4 --autorepair scaling
```
[the remark this answers is not in the document]
Comment (kami619) [reply doc/benchmark/modules/ROOT/pages/report/rosa-benchmark-key-results.adoc:56]: https://github.com/keycloak/keycloak-benchmark/issues/1002#issuecomment-2397820486

There was a discussion here, after my initial tests @mhajas.

File: provision/aws/rosa_create_cluster.sh [hunk 2]
```diff
@@ -63,7 +63,7 @@ fi
 
 SCALING_MACHINE_POOL=$(rosa list machinepools -c "${CLUSTER_NAME}" -o json | jq -r '.[] | select(.id == "scaling") | .id')
 if [[ "${SCALING_MACHINE_POOL}" != "scaling" ]]; then
-    rosa create machinepool -c "${CLUSTER_NAME}" --instance-type m5.4xlarge --max-replicas 10 --min-replicas 1 --name scaling --enable-autoscaling --autorepair
+    rosa create machinepool -c "${CLUSTER_NAME}" --instance-type m5.2xlarge --max-replicas 15 --min-replicas 1 --name scaling --enable-autoscaling --autorepair
```
[the remark this answers is not in the document]
Comment (kami619) [reply provision/aws/rosa_create_cluster.sh:66]: That is a good idea, do you want me to attempt it in this PR or a new task for it ?

File: doc/kubernetes/modules/ROOT/pages/prerequisite/prerequisite-rosa.adoc [hunk 3]
```diff
@@ -85,14 +85,14 @@ The above installation script creates an admin user automatically but in case th
 == Scaling the cluster's nodes on demand
 
 The standard setup of nodes might be too small for running a load test, at the same time using a different instance type and rebuilding the cluster takes a lot of time (about 45 minutes).
-To scale the cluster on demand, the standard setup has a machine pool named `scaling` with instances of type `m5.4xlarge` which is auto-scaled based on the current demand from 0 to 10 instances.
+To scale the cluster on demand, the standard setup has a machine pool named `scaling` with instances of type `m5.2xlarge` which is auto-scaled based on the current demand from 0 to 10 instances.
```
Comment (pruivo) [inline doc/kubernetes/modules/ROOT/pages/prerequisite/prerequisite-rosa.adoc:88]: Updated the number of instances. 
```suggestion
To scale the cluster on demand, the standard setup has a machine pool named `scaling` with instances of type `m5.2xlarge` which is auto-scaled based on the current demand from 4 to 15 instances.
```

File: doc/benchmark/modules/ROOT/pages/report/rosa-benchmark-key-results.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhajas) [reply doc/benchmark/modules/ROOT/pages/report/rosa-benchmark-key-results.adoc:56]: Ahh, sorry I missed that. Makes sense then.

File: doc/kubernetes/modules/ROOT/pages/prerequisite/prerequisite-rosa.adoc (hunk 3 above)
Comment (pruivo) [reply doc/kubernetes/modules/ROOT/pages/prerequisite/prerequisite-rosa.adoc]: @kami619 I'm unsure if the minimum is 1 or 4 at this point. It looks like `rosa_create_cluster.sh` sets the minimum to 1.

File: provision/aws/rosa_create_cluster.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhajas) [reply provision/aws/rosa_create_cluster.sh:66]: Hmm if it is as simple as this:
```suggestion
    rosa create machinepool -c "${CLUSTER_NAME}" --instance-type ${COMPUTE_MACHINE_TYPE:-m5.2xlarge} --max-replicas 15 --min-replicas 1 --name scaling --enable-autoscaling --autorepair
```

I would do it here. If I am missing something then we can choose to do it in some follow-up issue.

**Comment by kami619** [comment]
@mhajas applied the changes from the feedback. I will do a smoke test with this and see if it goes through on my fork, before we merge it.

**Review by pruivo — APPROVED** [review]

**Review by mhajas — APPROVED** [review]
Thanks @kami619

**Comment by kami619** [comment]
@mhajas the smoke test ran as expected. I am merging this PR.

**Pull request merged**